src/processing/aggregate_counts.py

```diff
@@ -0,0 +1,29 @@
+
+from pyspark.sql import functions as F
+
+def get_aggregate_counts(df, counting_col, alias_name):
+    '''
+    Function to get counts in dataframe 
+    '''
+    df = (df
+        .agg(
+            F.countDistinct(counting_col).alias(alias_name)
+        )
+    )
+
+    return df
+
+
+def get_grouped_aggregate_counts(df, grouping_col, counting_col, alias_name):
+    '''
+    Function to get counts in dataframe based on grouping
+    '''
+    df = (df
+        .groupBy(grouping_col)
+        .agg(
+            F.countDistinct(counting_col).alias(alias_name)
+        )
+        .orderBy(grouping_col)
+    )
```

Comment:
```suggestion
    df = (
        df
        .groupBy(grouping_col)
        .agg(F.countDistinct(counting_col).alias(alias_name))
        .orderBy(grouping_col)
    )
```

Reply:
I'm not a huge fan of reassigning a variable, so I would rename the two "df"s different things, e.g.:
```
def get_grouped_aggregate_counts(df_unaggregated, grouping_col, counting_col, alias_name):
    '''
    Function to get counts in dataframe based on grouping
    '''
    df_aggregated = (df_unaggregated
        .groupBy(grouping_col)
        .agg(
            F.countDistinct(counting_col).alias(alias_name)
        )
        .orderBy(grouping_col)
    )
	return df_aggregated

```
(Ignore any dodgy indentation etc. - this was purely to raise the idea of not reassigning "df".